src/connectors/meraki.py

```diff
@@ -0,0 +1,181 @@
+    ('TAGS', 'VARCHAR(256)'),
+    ('LNG', 'FLOAT'),
+    ('LAT', 'FLOAT'),
+]
+
+
+def get_data(url: str, token: str, params: dict = {}) -> dict:
+    headers: dict = {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "X-Cisco-Meraki-API-Key": f"{token}",
+    }
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        req = requests.get(url, params=params, headers=headers)
+        req.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise http_err
+    log.debug(req.status_code)
+    return req.json()
+
+
+def connect(connection_name, options):
+    landing_table_client = f'data.meraki_{connection_name}_connection_client'
+    landing_table_device = f'data.meraki_{connection_name}_connection_device'
+    options['network_id_whitelist'] = options.get('network_id_whitelist', '').split(',')
+
+    comment = yaml_dump(module='meraki', **options)
+
+    db.create_table(name=landing_table_client,
+                    cols=LANDING_TABLE_COLUMNS_CLIENT, comment=comment)
+    db.execute(f'GRANT INSERT, SELECT ON {landing_table_client} TO ROLE {SA_ROLE}')
+    db.create_table(name=landing_table_device,
+                    cols=LANDING_TABLE_COLUMNS_DEVICE, comment=comment)
+    db.execute(f'GRANT INSERT, SELECT ON {landing_table_device} TO ROLE {SA_ROLE}')
+    return {
+        'newStage': 'finalized',
+        'newMessage': "Meraki ingestion tables created!",
+    }
+
+
+def parse_number(value):
+    if value:
+        return value
+    return None
+
+
+def ingest(table_name_client, landing_table_device, options):
+    landing_table_client = f'data.{table_name_client}'
+    landing_table_device = f'data.{landing_table_device}'
+    timestamp = datetime.utcnow()
+    api_key = options['api_key']
+    whitelist = options['network_id_whitelist']
+
+    for network in whitelist:
+        
+        try:
+            devices = get_data(f"https://api.meraki.com/api/v0/networks/{network}/devices", api_key)
+        except requests.exceptions.HTTPError as e:
+            log.error(f"{network} not accessible, ")
+            log.error(e)
+            continue
+
+        for device in devices:
+            serial_number = device['serial']
+            clients = get_data(f"https://api.meraki.com/api/v0/devices/{serial_number}/clients", api_key)
+        
+            db.insert(
+                landing_table_client,
+                values=[(
+                    timestamp,
+                    client,
+                    client.get('id','None'),
+                    client.get('mac','None'),
+                    client.get('description','None'),
+                    client.get('mdnsName','None'),
+                    client.get('dhcpHostname','None'),
+                    client.get('ip','None'),
+                    parse_number(client.get('vlan','None')),
```

Comment:
Can't this just be `client.get('vlan', None)` instead of wrapping it in parse_number?

Reply:
The API is returning empty strings which then error out when trying to parse as a number

Reply:
Hmm, okay. I'll think about this a bit; I suspect there's a better way to handle this.

Reply:
not sure what the client value is, but agreed this looks buggy. right now it does:

    client.get('vlan', 'None') or None

are you sure this is what you want?